feat: Add Revenue Analytics v0

rafaeelaudibert opened the pull request:
This our first attempt at Revenue Analytics. This is just the initial setup, configuring all of the file structure and attempting at using a virtual `view` to display this inside PostHog. We arent adding any charts yet, this is coming soon.

We'll probably need to create cross-table views (for invoices and charges), but that's coming in a future PR. This is only the first PR to get an idea from the Data Warehouse team whether this makes any sense or not

Supersedes https://github.com/PostHog/posthog/pull/30704

rafaeelaudibert committed:
feat: Add Revenue Analytics v0

This our first attempt at Revenue Analytics. This is just the initial setup, configuring all of the file structure and attempting at using a virtual `view` to display this inside PostHog. We arent adding any charts yet, this is coming soon.

We'll probably need to create cross-table views (for invoices and charges) but that's coming in a future PR, this is only the first PR to get an idea from the Data Warehouse team whether this makes any sense or not

EDsCODE commented on products/revenue_analytics/backend/hogql_queries/revenue_example_data_warehouse_tables_query_runner.py:
```diff
@@ -1,77 +1,74 @@
+        for view_name in self.database.get_views():
+            view = self.database.get_table(view_name)
+            if isinstance(view, RevenueAnalyticsRevenueView):
+                view = cast(RevenueAnalyticsRevenueView, view)
```
I think this cast is unnecessary now?

rafaeelaudibert replied:
I'll check. I think it's still needed because mypy is not very smart

EDsCODE reviewed (approved):
**Looked at the hogql/data warehouse querying code. This looks better and easier to work into materializations now

rafaeelaudibert committed:
tests: Simplify tests

Most werent necessary anymore, we've simplified the workflow by a lot now

github-actions[bot] committed:
Update UI snapshots for `chromium` (3)

Update query snapshots

Update query snapshots

Update UI snapshots for `chromium` (5)

Update UI snapshots for `chromium` (5)

Update UI snapshots for `chromium` (5)

Update UI snapshots for `chromium` (5)

Update UI snapshots for `chromium` (5)

rafaeelaudibert committed:
test: Fix broken test

There's one extra query now to check for Stripe sources

rafaeelaudibert committed:
fix: Remove N+1 queries

Turns out increasing the number of queries in the test file was wrong, let's remove all of the N+1 queries

lricoy commented on tach.toml:
```diff
@@ -39,15 +25,21 @@ depends_on = [
+    # NOTE: Add new product dependencies here and in settings/web.py PRODUCTS_APPS
     "products.early_access_features",
+    "products.editor",
+    "products.revenue_analytics",
```
Just a note to check if the style changes are due to a local or different formatter; nothing is required on this PR.

rafaeelaudibert committed:
test: Fix tests

rafaeelaudibert committed:
fix: Fix type casts

github-actions[bot] committed:
Update UI snapshots for `chromium` (5)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

rafaeelaudibert committed:
test: Fix flaky
